docs/content/operating.html.md

```diff
@@ -0,0 +1,233 @@
+necessary.
+
+Each host should have a minimum of 1GB of memory, and inter-host network packets
+should have a latency of less than 2ms. Deploying a single Flynn cluster across
+higher latency WAN links is not recommended, as it can have a negative impact on
+cluster consensus.
+
+## Storage
+
+Flynn uses ZFS to store data. By default, a ZFS pool is created in a sparse file
+on the existing filesystem. This is not recommended for production for
+performance and reliability reasons. Before starting the `flynn-host` daemon,
+you can create a ZFS pool named `flynn-default` and it will be used instead of
+a sparse file.
+
+```text
+$ zpool create flynn-default /dev/sdb1
+```
+
+If you already have a Flynn cluster running, you can move the existing ZFS pool
+off of the sparse file by first attaching your disk as a mirror, then detaching
+the sparse file after it has been replicated to the new disk:
+
+```text
+# Attach /dev/sdb1 to the flynn-default ZFS pool
+$ sudo zpool attach flynn-default /var/lib/flynn/volumes/zfs/vdev/flynn-default-zpool.vdev /dev/sdb1
+
+# Check the replication status
+$ sudo zpool status flynn-default
+  pool: flynn-default
+ state: ONLINE
+  scan: resilvered 59.7M in 0h0m with 0 errors on Mon Nov  2 04:02:58 2015
+config:
+
+	NAME                                                          STATE     READ WRITE CKSUM
+	flynn-default                                                 ONLINE       0     0     0
+	  mirror-0                                                    ONLINE       0     0     0
+	    /var/lib/flynn/volumes/zfs/vdev/flynn-default-zpool.vdev  ONLINE       0     0     0
+	    sdb1                                                      ONLINE       0     0     0
+
+errors: No known data errors
+
+# Detach the sparse file from the ZFS pool and delete it
+$ sudo zpool detach flynn-default /var/lib/flynn/volumes/zfs/vdev/flynn-default-zpool.vdev
+$ sudo rm /var/lib/flynn/volumes/zfs/vdev/flynn-default-zpool.vdev
+```
+
+## DNS and Load Balancing
+
+Flynn has a built-in router that handles all incoming HTTP, HTTPS and TCP
+traffic. It is lightweight and runs on every Flynn host, this allows easy
+deployment of Flynn without requiring another load balancer in front of Flynn.
+
+We recommend running Flynn with low-TTL round-robin DNS A records pointing at
+each host. For automatic failover, health checks should be configured that
+automatically remove a host's A record if it is unhealthy.
+
+Alternatively, a TCP load balancer for ports 80 and 443 can be configured in
+front of the Flynn router, this may increase overhead and complexity but can
+make sense in some environments.
+
+## Firewalling
+
+A firewall preventing external access must always be configured on or in front
+of Flynn hosts, only these ports should be allowed:
+
+* 80 (HTTP)
+* 443 (HTTPS)
+* 3000 to 3500 (user-defined TCP services, optional)
+
+Internal cross-host cluster communication happens on a variety of ports and
+should not be restricted.
```

Comment:
I think it is worth explicitly mentioning that both TCP & UDP should be unrestricted (I have seen users just allowing TCP).